cbmc/proofs/IotMqtt_PublishAsync/IotMqtt_PublishAsync_harness.c

```diff
@@ -0,0 +1,79 @@
+#include "iot_config.h"
+#include "private/iot_mqtt_internal.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "mqtt_state.h"
+
+/**
+ * We abstract all functions related to concurrency and assume they are correct.
+ * Thus, we add these stub to increase coverage.
+ */
+IotTaskPoolError_t IotTaskPool_CreateJob( IotTaskPoolRoutine_t userCallback,
+                                          void * pUserContext,
+                                          IotTaskPoolJobStorage_t * const pJobStorage,
+                                          IotTaskPoolJob_t * const pJob )
```

Comment:
This function also act if the parameters are NULL. It checks them with `TASKPOOL_ON_NULL_ARG_GOTO_CLEANUP`, which exit if an argument is NULL.

Reply:
Yes, good catch! In this case, the function `_IotMqtt_ScheduleOperation` asserts that `IotTaskPool_CreateJob` will always return `IOT_TASKPOOL_SUCCESS`. If ever a code change would lead to invalid parameters in this function, not only the assertion would fail, but the proof as well.